akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -43,12 +43,10 @@ private[akkastream] class DocumentStage[T](
       @inline private def tailable = cursor.wrappee.tailable
 
       private var request: () => Future[Option[Response]] = { () =>
+        // Call onFirst immediately to make sure only one request is sent.
         cursor.makeRequest(maxDocs).andThen {
-          case Success(r) if (
-            tailable && r.reply.numberReturned > 0
-          ) => onFirst()
-
-          case Success(_) if (!tailable) => onFirst()
+          case Success(_) =>
```

Comment:
I think it's reasonable for tailable cursors.  The [official document ](https://docs.mongodb.com/manual/core/tailable-cursors/) explains the reason:
> Because tailable cursors do not use indexes, the initial scan for the query may be expensive.